src/events.js

```diff
@@ -1919,19 +1940,58 @@ export const goToParticipantSearch = () => {
     document.getElementById('navBarSearch').click();
 }
 
-export const addEventSpecimenLinkForm = (formData) => {
+export const addEventSpecimenLinkForm = (formData, visitConcept) => {
     const form = document.getElementById('specimenLinkForm');
     const connectId = document.getElementById('specimenContinue').dataset.connectId;
 
     if (document.getElementById('navBarParticipantCheckIn')) document.getElementById('navBarParticipantCheckIn').dataset.connectId = connectId;
 
-    form.addEventListener('submit', e => {
+    form.addEventListener('submit', async (e) => {
         e.preventDefault();
-        btnsClicked(connectId, formData);
+        const query = `connectId=${parseInt(connectId)}`;
+        const participant  = await findParticipant(query);
+        const data = participant.data[0];
```

Comment:
can this break if participant is null?

Reply:
I followed the same logic found in several other instances. See some examples below:
https://github.com/episphere/biospecimen/blob/876dd95a79c57db541d48119d071f3ac92fa2a1a/src/events.js#L2561

https://github.com/episphere/biospecimen/blob/876dd95a79c57db541d48119d071f3ac92fa2a1a/src/events.js#L139

https://github.com/episphere/biospecimen/blob/876dd95a79c57db541d48119d071f3ac92fa2a1a/src/events.js#L1790

https://github.com/episphere/biospecimen/blob/876dd95a79c57db541d48119d071f3ac92fa2a1a/src/events.js#L1820)

Reply:
So for example, in the first instance shared above (i.e in function addEventReturnToCollectProcess), data variable is passed in tubeCollectedTemplate and I don't see any null check/validation in that function, so I am assuming that findParticipant won't give null response.